plugins/azure-devops-backend/src/service/router.test.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { ConfigReader } from '@backstage/config';
+import express from 'express';
+import request from 'supertest';
+import { AzureDevOpsApi } from '../api';
+import { createRouter } from './router';
+import { RepoBuild } from '../api/types';
+import { GitRepository } from 'azure-devops-node-api/interfaces/GitInterfaces';
+import {
+  Build,
+  BuildResult,
+  BuildStatus,
+} from 'azure-devops-node-api/interfaces/BuildInterfaces';
+
+describe('createRouter', () => {
+  let azureDevOpsApi: jest.Mocked<AzureDevOpsApi>;
+  let app: express.Express;
+
+  beforeAll(async () => {
+    azureDevOpsApi = {
+      getGitRepository: jest.fn(),
+      getBuildList: jest.fn(),
+      getRepoBuilds: jest.fn(),
+    } as any;
+    const router = await createRouter({
+      azureDevOpsApi,
+      logger: getVoidLogger(),
+      config: new ConfigReader({
+        azureDevOps: {
+          token: 'foo',
+          host: 'host.com',
+          organization: 'myOrg',
+          top: 5,
+        },
+      }),
+    });
+    app = express().use(router);
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('GET /health', () => {
+    it('returns ok', async () => {
+      const response = await request(app).get('/health');
+
+      expect(response.status).toEqual(200);
+      expect(response.body).toEqual({
+        status: 'Healthy',
+        details: 'All required config has been provided',
+      });
+    });
+  });
+
+  describe('GET /repository/:projectName/:repoName', () => {
+    it('fetches a single repository', async () => {
+      const gitRepository: GitRepository = {
+        id: 'af4ae3af-e747-4129-9bbc-d1329f6b0998',
+        name: 'myRepo',
+        url: 'https://host.com/repo',
+        defaultBranch: 'refs/heads/develop',
+        sshUrl: 'ssh://host.com/repo',
+        webUrl: 'https://host.com/webRepo',
+      };
+
+      azureDevOpsApi.getGitRepository.mockResolvedValueOnce(gitRepository);
+
+      const response = await request(app).get(
+        `/repository/${'myProject'}/${'myRepo'}`,
+      );
+
+      expect(response.status).toEqual(200);
+      expect(response.body).toEqual(gitRepository);
+    });
+  });
+
+  describe('GET /builds/:projectName/:repoId', () => {
```

Comment:
repoId here, repoName above - probably not important, but still an inconsistency

Reply:
The way the Azure DevOps API is setup you can't get the Builds for a repo using the repo name, you need to get its Id. I didn't feel like that would be a smooth experience on the annotation side having to find the Id. Does that make sense?